mart/attack/perturber.py

```diff
@@ -90,18 +96,25 @@ def __init__(
 
         self.perturbation = None
 
-    def configure_perturbation(self, input: torch.Tensor | tuple | tuple[dict[str, torch.Tensor]]):
+    def configure_perturbation(
+        self,
+        input: torch.Tensor | tuple[torch.Tensor] | tuple[dict[str, torch.Tensor]],
+        target: torch.Tensor | tuple[dict[str, Any]],
+    ):
         def create_and_initialize(data, *, input, target, modality):
-            # Though data and target are not used, they are required placeholders for modality_dispatch().
-            # TODO: we don't want an integer tensor, but make sure it does not affect mixed precision training.
-            pert = torch.empty_like(input, dtype=torch.float, requires_grad=True)
+            # Though data is not used, it is a required placeholder for modality_dispatch().
+            pert = self.creator[modality](input=input, target=target)
```

Comment:
I don't think this is useful because a perturbation can't have more pixels than are in the input. Unless this is meant to optimize the perturbation size? In that case, I think this optimization is premature.